contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -2320,6 +2323,95 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 
 	switch (nodeTag(parsetree))
 	{
+		case T_AlterFunctionStmt:
+			{
+				/*
+				* For ALTER PROC, we will:
+				* 1. Save important pg_proc metadata from the current proc (oid, proacl)
+				* 2. drop the current proc
+				* 3. create the new proc
+				* 4. update the pg_proc entry for the new proc with metadata from the old proc
+				* 5. update the babelfish_function_ext entry for the existing proc with new metadata based on the new proc
+				*/
+				AlterFunctionStmt *stmt = (AlterFunctionStmt *) parsetree;
+				bool 				isCompleteQuery = (context != PROCESS_UTILITY_SUBCOMMAND);
+				bool 				needCleanup;
+				Oid					oldoid;
+				Acl					*proacl;
+				bool				isSameProc;
+				ObjectAddress 		address;
+				CreateFunctionStmt	*cfs;
+				ListCell 			*option, *location_cell = NULL;
+				int 				origname_location = -1;
+
+				if (stmt->objtype != OBJECT_PROCEDURE)
+					break;
+
+				/* All event trigger calls are done only when isCompleteQuery is true */
+				needCleanup = isCompleteQuery && EventTriggerBeginCompleteQuery();
+
+				/* PG_TRY block is to ensure we call EventTriggerEndCompleteQuery */
+				PG_TRY();
+				{
+					StartTransactionCommand();
+					if (isCompleteQuery)
+						EventTriggerDDLCommandStart(parsetree);
+
+					foreach (option, stmt->actions)
+					{
+						DefElem *defel = (DefElem *) lfirst(option);
+						if (strcmp(defel->defname, "location") == 0)
+						{
+							/*
+							* location is an implicit option in tsql dialect,
+							* we use this mechanism to store location of function
+							* name so that we can extract original input function
+							* name from queryString.
+							*/
+							origname_location = intVal((Node *) defel->arg);
+							location_cell = option;
+							pfree(defel);
+						}
+					}
+
+					/* delete location cell if it exists as it is for internal use only */
+					if (location_cell)
+						stmt->actions = list_delete_cell(stmt->actions, location_cell);
+
+					/* make a CreateFunctionStmt to pass into CreateFunction() */
+					cfs = makeNode(CreateFunctionStmt);
+					cfs->is_procedure = true;
+					cfs->replace = true;
+					cfs->funcname = stmt->func->objname;
+					cfs->parameters = stmt->func->objfuncargs;
+					cfs->returnType = NULL;
+					cfs->options = stmt->actions;
+
+					pltsql_proc_get_oid_proname_proacl(stmt, pstate, &oldoid, &proacl, &isSameProc);
+					if (!isSameProc) /* i.e. different signature */
+						RemoveFunctionById(oldoid);
+					address = CreateFunction(pstate, cfs); // if this is the same proc, will just update the existing one
+					pg_proc_update_oid_acl(address, oldoid, *proacl);
+					/* Update function/procedure related metadata in babelfish catalog */
+					if (isSameProc)
+						pltsql_store_func_default_positions(address, cfs->parameters, queryString, origname_location);
```

Comment:
can you add a comment here explaining why you don't need to do all of the extra catalog modifications in this case?

Reply:
Added a comment to this line

Reply:
I think the comment needs more detail. In particular, we should explain that in the case that the signature is the same, then no other catalog entries are made, in pg_depend, or other places, due to the fact that CreateFunction just modifies the existing function instead of creating a new one. On the other hand, if the signature is different, then there are dangling dependencies from the original function that need to be cleaned up since new ones get created for the new function signature. Consider updating the comment in the `else` clause to help clarify the logic being used here.